pkg/util/logutil/log.go

```diff
@@ -395,3 +396,25 @@ func proxyFields() []zap.Field {
 	}
 	return fields
 }
+
+// SampleLoggerFactory returns a factory function that creates a sample logger.
+// the logger is used to sample the log to avoid too many logs, it will only log
+// the first 'first' log entries with the same level and message in 'tick' time.
+// NOTE: Because we need to record the log count with the same level and message
+// in this specific logger, the returned factory function will only create one logger.
+// this logger support at most 4096 types of logs with the same level and message.
+func SampleLoggerFactory(tick time.Duration, first int, fields ...zap.Field) func() *zap.Logger {
```

Comment:
conflicts with https://github.com/pingcap/tidb/pull/54438

Reply:
I hope git can automatic merge the commit with same file diff